### ISSUE TITLE
Validation step is not happening while training

### ISSUE THREAD

**Issue by shekarneo** [ultralytics/yolov5#7397]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

Hi Run the training script with 5000 epochs and save weights at every 50 epochs, but it is not doing validation step and not giving any metric results.

```
python3 train.py --data ../training/dataset.yaml --weights '' --cfg models/yolov5l.yaml --cache --img 640 --evolve --batch-size 8 --save-period 50 --epochs 5000 --optimizer Adam
```

its been 12th epochs getting these results:

  Epoch   gpu_mem       box       obj       cls    labels  img_size
    8/4999     5.04G   0.06438   0.08565   0.01406        51       640: 100%|██████████| 1435/1435 [05:42<00:00,  4.20it/s]                           

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    9/4999     5.04G   0.06397   0.08573   0.01407        42       640: 100%|██████████| 1435/1435 [05:51<00:00,  4.09it/s]                           

     Epoch   gpu_mem       box       obj       cls    labels  img_size
   10/4999     5.04G   0.06388   0.08541   0.01401        21       640: 100%|██████████| 1435/1435 [05:48<00:00,  4.12it/s]                           

     Epoch   gpu_mem       box       obj       cls    labels  img_size
   11/4999     5.04G   0.06371   0.08526   0.01384        32       640: 100%|██████████| 1435/1435 [05:49<00:00,  4.11it/s]  

### Minimal Reproducible Example

```
python3 train.py --data ../training/dataset.yaml --weights '' --cfg models/yolov5l.yaml --cache --img 640 --evolve --batch-size 8 --save-period 50 --epochs 5000 --optimizer Adam
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@shekarneo there's no bug, it just seems you don't understand how evolution works. Evolution with `--evolve` argument does not output any intermediate metrics nor save any models, it outputs a set of optimized hyperparameters after several weeks of training hundreds of models. See Hyperparameter Evolution tutorial for details.

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️  RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Roboflow for Datasets, Labeling, and Active Learning](https://docs.ultralytics.com/yolov5/tutorials/roboflow_datasets_integration)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [Architecture Summary](https://docs.ultralytics.com/yolov5/tutorials/architecture_description)&nbsp; ⭐ NEW

Good luck 🍀 and let us know if you have any other questions!

**Comment by shekarneo** [comment]
Thanks @glenn-jocher, how do i monitor precesion and recall.

**Comment by glenn-jocher** [comment]
@shekarneo Precision and recall can be monitored using the integrated W&B logging feature in YOLOv5. W&B automatically logs and visualizes all model metrics including precision and recall. Simply enable the `--project` and `--name` arguments to start logging your metrics. For further guidance, please refer to the Weights & Biases Logging tutorial in the YOLOv5 documentation.

If you have any more questions feel free to ask!
